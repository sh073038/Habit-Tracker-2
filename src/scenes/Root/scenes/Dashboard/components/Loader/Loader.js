// @flow

import * as React from 'react'
import ContentLoader from 'react-content-loader'

type Props = {
  windowWidth: number,
}

const Loader = ({ windowWidth }: Props) => (
  <ContentLoader height={480}>
    {windowWidth >= 600 ? (
      <React.Fragment>
        <circle cx="20" cy="25" r="10" />
        <rect x="35" y="18" rx="4" ry="4" width="85" height="14" />
        <rect x="10" y="50" rx="5" ry="5" width="400" height="200" />
        <rect x="10" y="260" rx="5" ry="5" width="400" height="150" />
      </React.Fragment>
    ) : (
      <React.Fragment>
        <circle cx="30" cy="25" r="15" />
        <rect x="55" y="18" rx="4" ry="4" width="85" height="14" />
        <rect x="10" y="50" rx="5" ry="5" width="400" height="450" />
      </React.Fragment>
    )}
  </ContentLoader>
)

export default Loader
