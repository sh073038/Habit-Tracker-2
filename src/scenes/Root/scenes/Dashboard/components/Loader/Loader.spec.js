import React from 'react'
import { shallow } from 'enzyme'

import Loader from './Loader'

describe('Loader', () => {
  it('renders component', () => {
    const wrapper = shallow(<Loader />)
    expect(wrapper.getElement()).toMatchSnapshot()
  })
})
