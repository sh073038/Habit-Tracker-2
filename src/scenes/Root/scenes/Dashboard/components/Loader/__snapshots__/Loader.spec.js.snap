// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders component 1`] = `
<ContentLoader
  height={480}
>
  <UNDEFINED>
    <circle
      cx="30"
      cy="25"
      r="15"
    />
    <rect
      height="14"
      rx="4"
      ry="4"
      width="85"
      x="55"
      y="18"
    />
    <rect
      height="450"
      rx="5"
      ry="5"
      width="400"
      x="10"
      y="50"
    />
  </UNDEFINED>
</ContentLoader>
`;
